packages/algob/src/internal/deployer.ts

```diff
@@ -526,7 +526,12 @@ class DeployerBasicMode {
 				break;
 			}
 			case wtypes.SignType.LogicSignature: {
-				signer.lsig.lsig.args = signer.args ?? [];
+				if (signer?.lsig?.lsig?.args) {
+					signer.lsig.lsig.args = signer.args ?? [];
+				} else {
+					(signer.lsig as any).args = signer.args ?? []; // args property didn't exist in earlier version of API (for reference: see the multisig example)
```

Comment:
we should avoid as any whenever possible

Reply:
Actually it is necessary since the new type of LogicSigAccount type consist of args, due to which it was giving me a type error which casting it to any